File: app/src/release/java/com/mycompany/myapp/monitoring/LoggerModule.java
package com.mycompany.myapp.monitoring;

import com.mycompany.myapp.monitoring.model.NoOpTree;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import timber.log.Timber.Tree;

@Module
public class LoggerModule {
    @Singleton
    @Provides
    public Tree provideTimberTree() {
        return new NoOpTree();
    }
}

File: app/src/release/java/com/mycompany/myapp/monitoring/LoggerModule.kt
package com.mycompany.myapp.monitoring

import com.mycompany.myapp.monitoring.model.NoOpTree
import dagger.Module
import dagger.Provides
import timber.log.Timber
import javax.inject.Singleton

@Module
class LoggerModule {
    @Singleton
    @Provides
    fun provideTimberTree() : Timber.Tree = NoOpTree()
}
